### PR TITLE
Feature/marketplace settlement

### PR DESCRIPTION
**Pull request by prashantesmagico**
PR for the Settlement requests

### CODE REVIEW

File: src/app.module.ts [hunk 1]
```diff
@@ -12,7 +12,7 @@ import { RequestModule } from './request/request.module';
       isGlobal: true, // Make the configuration available globally
     }),
     PrismaModule,
-    RequestModule,
+    SettlementModule,
```
Comment (VamshiBatta07) [inline src/app.module.ts:15]: Do not remove the Request Module.

**Comment by VamshiBatta07** [comment]
- [x] Needs test cases.

File: src/settlement/dto/create-settlement.dto.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+  requestStatus: SettlementStatusEnum;
+}
+
+// SettlementStatusDto is used for updating the status of a request.
+export class thirdPartyResponseStatusEnumDto {
+  // New status for the request, should be one of the valid UserResponseStatusEnum values.
+  @IsNotEmpty()
+  @IsEnum(thirdPartyResponseStatusEnum)
+  thirdPartyResponseStatus: thirdPartyResponseStatusEnum;
+}
```
Comment (VamshiBatta07) [inline src/settlement/dto/create-settlement.dto.ts:79]: If there is no future use case for "SettlementStatusDto" and "thirdPartyResponseStatusEnumDto" please remove them as they are not used anywhere.

File: src/settlement/settlement.service.ts
Comment (VamshiBatta07) [inline src/settlement/settlement.service.ts:1]: Needs error handling across all APIs.

- [x] createSettlement - for duplicate entries.
- [x] getAllSettlementForUser - if userId not found.
- [x] getsettlementById - if requestId not found.
- [x] updateSettlementByRequestId - if requestId not found.

File: src/settlement/entities/settlement.entity.ts
Comment (VamshiBatta07) [inline src/settlement/entities/settlement.entity.ts:1]: Remove this if not necessary.

File: src/settlement/dto/create-settlement.dto.ts (hunk 2 above)
Comment (VamshiBatta07) [inline src/settlement/dto/create-settlement.dto.ts:79]: If there is no future use case for "SettlementStatusDto" and "thirdPartyResponseStatusEnumDto" please remove them as they are not used anywhere.

File: src/app.module.ts (hunk 1 above)
Comment (VamshiBatta07) [inline src/app.module.ts:15]: Do not remove the Request Module.

**Comment by VamshiBatta07** [comment]
Just a suggestion.
Add an index.ts file for the dto-folder and import all the dtos into it.
Import the dtos from dto/index.ts when necessary. 
It will look good.

File: src/settlement/entities/settlement.entity.ts
Comment (prashantesmagico) [reply src/settlement/entities/settlement.entity.ts:1]: Removed

File: src/settlement/dto/create-settlement.dto.ts (hunk 2 above)
Comment (prashantesmagico) [reply src/settlement/dto/create-settlement.dto.ts:79]: Removed

**Comment by Ajay-Maury** [comment]
please go with PR :- https://github.com/COMPASS-DPG/marketplace-request-service/pull/5

File: src/app.module.ts (hunk 1 above)
Comment (prashantesmagico) [reply src/app.module.ts:15]: Done

File: src/settlement/settlement.service.ts
Comment (prashantesmagico) [reply src/settlement/settlement.service.ts:1]: Added error handler across the API

**Pull request closed without merging**